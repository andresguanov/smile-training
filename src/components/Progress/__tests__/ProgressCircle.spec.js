import { shallowMount } from "@vue/test-utils"
import smProgressCircle from "@components/Progress/ProgressCircle"
let wrapper

describe("smProgressCircle", () => {
  //#region Basic render
  it("renders default smProgressCircle", () => {
    wrapper = shallowMount(smProgressCircle)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toBe("")
    expect(wrapper.classes("show-percentage")).toBe(false)
  })
  //#endregion

  //#region Props
  it("shows percentage", () => {
    wrapper = shallowMount(smProgressCircle, {
      propsData: {
        value: 20,
        max: 200,
        showPercentage: true
      }
    })
    expect(wrapper.attributes("data-percentage")).toBe("10")
  })
  //#endregion

  //#region Methods
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  //#endregion
})
