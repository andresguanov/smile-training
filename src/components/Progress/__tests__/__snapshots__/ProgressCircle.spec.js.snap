// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smProgressCircle renders default smProgressCircle 1`] = `<div class="progress-circle is-medium"></div>`;
